### ISSUE TITLE
Conflict with Prompt fusion extension

### ISSUE THREAD

**Issue by asagi4** [ljleb/sd-webui-neutral-prompt#31]
There seems to be a conflict between prompt fusion and this extension when using variables, even when NP is not in use.

I'm on mobile right now, so I can't give you all the details, but I did disable most of my extensions that may have conflicted. This should be easy to reproduce with a simple prompt:
```
$p=cat, [blue fur:red fur:,]
$p
```

With Neutral Prompt enabled, it seems this gets interpreted as an empty prompt. If I don't use variables, things work as expected.

**Comment by ljleb** [comment]
Indeed, there is a situation here. Neutral prompt transforms newlines into whitespace before further processing. The reason newlines are transformed is that the webui prompt parser stops handling AND keywords properly as soon as there is a newline anywhere other than in the first prompt.

There are multiple options here. I could remove variables from PF because there are other extensions that add these refactoring capabilities.

I'm not sure the AND parsing is easily monkey patched, so I could open a PR in the webui. Once it is handled properly the extension will be able to work without stripping newlines.

**Comment by asagi4** [comment]
I tried this, but it doesn't seem to work quite right. As far as I can tell, neutral prompt still sees the prompt without the variables expanded because it hijacks a function that gets called before prompt fusion does variable expansion.

I tried to make NP hijack get_learned_conditioning and making sure the functions are called in order with prompt fusion going first, but that doesn't seem to be the correct solution either.
